extensions/sql/asset/index/src/test/java/org/eclipse/dataspaceconnector/sql/asset/index/SqlAssetIndexTest.java

```diff
@@ -0,0 +1,260 @@
+/*
+ *  Copyright (c) 2022 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial Tests
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.asset.index;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.dataloading.AssetEntry;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.monitor.ConsoleMonitor;
+import org.eclipse.dataspaceconnector.spi.persistence.EdcPersistenceException;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.transaction.datasource.DataSourceRegistry;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.sql.SqlQueryExecutor;
+import org.eclipse.dataspaceconnector.sql.datasource.ConnectionFactoryDataSource;
+import org.eclipse.dataspaceconnector.sql.datasource.ConnectionPoolDataSource;
+import org.eclipse.dataspaceconnector.sql.pool.ConnectionPool;
+import org.eclipse.dataspaceconnector.sql.pool.commons.CommonsConnectionPool;
+import org.eclipse.dataspaceconnector.sql.pool.commons.CommonsConnectionPoolConfig;
+import org.eclipse.dataspaceconnector.transaction.local.DataSourceResource;
+import org.eclipse.dataspaceconnector.transaction.local.LocalDataSourceRegistry;
+import org.eclipse.dataspaceconnector.transaction.local.LocalTransactionContext;
+import org.h2.jdbcx.JdbcDataSource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class SqlAssetIndexTest {
+
+    private static final String DATASOURCE_NAME = "asset";
+
+    private SqlAssetIndex sqlAssetIndex;
+    private ConnectionPool connectionPool;
+
+    @BeforeEach
+    void setUp() throws SQLException {
+        var monitor = new ConsoleMonitor();
```

Comment:
do not use console monitor, use either `new Monitor(){}` or `mock(Monitor.class)`